### PR TITLE
ADD: Mailcatcher + tests and docs

### PR DESCRIPTION
**Pull request by brettt89**

### CODE REVIEW

File: docs/07_environment_specs.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+### Sera
+
+Sera is used to lock a command to a specific server to avoid the same command running on multiple servers at the same time. While this is not needed for this environment specifically, it is used in SilverStripe Platform and so has been added to encourage use.
+
+**Service:**
+ - `/usr/local/bin/sera 0 php /var/www/mysite/www/framework/cli-script.php dev/tasks/<task>`
+
+### MSFonts
+
+MSFonts have been installed as these are included in SilverStripe platform releases.
```
Comment (Firesphere) [inline docs/07_environment_specs.md]: 👍

File: docs/07_environment_specs.md [hunk 2]
```diff
@@ -59,6 +59,12 @@ Sury's PHP 5.6 and PHP 7.1 have both been installed on this environment. You can
 
 **Logs:** `/var/log/syslog`
 **CLI:** `php framework/cli-script.php dev/tasks`
+**Switching Between Versions:**
+```bash
+update-alternatives --set php /usr/bin/php<5.6|7.1>
+update-alternatives --set php-config /usr/bin/php-config<5.6|7.1>
+update-alternatives --set phpize /usr/bin/phpize<5.6|7.1>
+```
```
Comment (Firesphere) [inline docs/07_environment_specs.md]: Would be nice to have it on the box, like the file watcher I'm using. But that's of later matter :)

File: docs/07_environment_specs.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+## Installed Applications
+ - **Nginx**
+ - **Apache2**
+ - **PHP**
+ - **MySQL**
+ - **Mailcatcher**
+ - **NVM**
+ - **node**
+ - **sera**
+ - **msfonts**
```
Comment (Firesphere) [inline docs/07_environment_specs.md]: Could you explicitly name the versions?

File: docs/07_environment_specs.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
+# Environment Specifications
+
+Below is a quick description of all services available in the final build. This build is based on the latest version of Debian (Jessie) which is currently 8.9
```
Comment (Firesphere) [inline docs/07_environment_specs.md]: Technically... the latest is Stretch ;)

Comment (brettt89) [reply docs/07_environment_specs.md]: I was meaning the latest version of Jessie lol

File: docs/07_environment_specs.md (hunk 3 above)
Comment (brettt89) [reply docs/07_environment_specs.md]: Can do, want to try and find a way to automate this.

File: docs/07_environment_specs.md (hunk 4 above)
Comment (Firesphere) [reply docs/07_environment_specs.md]: Easy confusion :D

File: docs/07_environment_specs.md (hunk 3 above)
Comment (Firesphere) [reply docs/07_environment_specs.md]: Ow yeah, automated would be frikkin' awesome!

File: spec/default/php_spec.rb [hunk 5]
```diff
@@ -65,4 +65,8 @@
   context php_config('memory_limit') do
     its(:value) { should eq "1024M" }
   end
+
+  context php_config('sendmail_path') do
+    its(:value) { should eq "/usr/bin/env catchmail" }
+  end
```
Comment (Firesphere) [inline spec/default/php_spec.rb]: I'm only missing a test for MailCatcher actually being online?

Comment (Firesphere) [reply spec/default/php_spec.rb]: Nevermind, I overlooked the tests in common. 🤦‍♂️

**Pull request merged**